client/wildcard/src/components/Popover/Popover.tsx

```diff
@@ -0,0 +1,181 @@
+import classNames from 'classnames'
+import { noop } from 'lodash'
+import React, {
+    createContext,
+    forwardRef,
+    MutableRefObject,
+    useCallback,
+    useContext,
+    useMemo,
+    useRef,
+    useState,
+} from 'react'
+import FocusLock from 'react-focus-lock'
+import { useCallbackRef, useMergeRefs } from 'use-callback-ref'
+
+import { useOnClickOutside, useKeyboard } from '../../hooks'
+import { ForwardReferenceComponent } from '../../types'
+
+import { FloatingPanel, FloatingPanelProps } from './floating-panel/FloatingPanel'
+
+enum PopoverOpenEventReason {
+    TriggerClick = 'TriggerClick',
+    TriggerFocus = 'TriggerFocus',
+    TriggerBlur = 'TriggerBlur',
+    ClickOutside = 'ClickOutside',
+    Esc = 'Esc',
+}
+
+interface PopoverOpenEvent {
+    isOpen: boolean
+    reason: PopoverOpenEventReason
+}
+
+interface PopoverContextData {
+    isOpen: boolean
+    targetElement: HTMLElement | null
+    tailElement: HTMLElement | null
+    anchor?: MutableRefObject<HTMLElement | null>
+    setOpen: (event: PopoverOpenEvent) => void
+    setTargetElement: (element: HTMLElement | null) => void
+    setTailElement: (element: HTMLElement | null) => void
+}
+
+const DEFAULT_CONTEXT_VALUE: PopoverContextData = {
+    isOpen: false,
+    targetElement: null,
+    tailElement: null,
+    setOpen: noop,
+    setTargetElement: noop,
+    setTailElement: noop,
+}
+
+const PopoverContext = createContext<PopoverContextData>(DEFAULT_CONTEXT_VALUE)
+
+interface PopoverProps {
+    anchor?: MutableRefObject<HTMLElement | null>
+    open?: boolean
```

Comment:
Does it make sense to use the same boolean prefix as we do in the context? 
```suggestion
    isOpen?: boolean
```